### ISSUE TITLE
intra-doc-link resolution failures should not be reported for cross-crate re-exports

### ISSUE THREAD

**Issue by euclio** [rust-lang/rust#56922]
Discovered in #56693.

Currently, if a crate re-exports an item from another crate with resolution failures, then those resolution failures get reported for the original crate. Since the downstream crate has no way to fix the resolution failure in the upstream crate, it shouldn't be reported, much like capping lints.

As of this writing, `std::io::Error` has an intra-link resolution failure in one of its impls. So, the following code triggers the issue.

```
use std::io::Error;
```

A fix for this issue should take care to still report failures for re-exports within the same crate.

cc #43466

**Comment by Assaf1978-boop** [comment]
Translation from Hebrew ,to English ,a
 and English,translation!!!
Assaf,Rosenkrantz.!!!

**Comment by jyn514** [comment]
Interesting, I was about to implement the opposite in https://github.com/rust-lang/rust/pull/73101. I assume you'd still want the warning for additional documentation on the re-export?
```rust
// crate a

// this will emit a warning when documenting a
/// [`BrokenLinkA`]
pub struct S;

// crate b

// this will give a warning for `BrokenLinkB`, but not for `BrokenLinkA`
/// Additional context: [`BrokenLinkB`]
pub use a::S;
```

**Comment by euclio** [comment]
Yes, that seems reasonable to me. The goal here is to avoid reporting warnings to downstream crates that have no way to fix them.

**Comment by jyn514** [comment]
A possible reason not to do this: it's possible for links to fail only when re-exported, even without bugs in rustdoc: https://github.com/rust-lang/rust/pull/75855

**Comment by jyn514** [comment]
It turns out this was already implemented and has a test case: https://github.com/rust-lang/rust/blob/master/src/test/rustdoc/intra-link-extern-crate.rs. I'm still not sure it makes sense to do this but I don't feel super strongly about it.
